### PR TITLE
[AWS] Add Route 53 Public Zone Datastream

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds a new datastream for Route 53 Public Zone logs

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by legoguy1000** [comment]
@kaiyan-sheng when u get a chance, could u take a look at this? I will fix the conflicts later today. I plan to create another PR for the private route 53 logging after this is merged.

**Comment by kaiyan-sheng** [comment]
@legoguy1000 Will do! Thanks for adding this and also using `aws-cloudwatch` input. I will read into the PR a bit more after lunch, just a quick note here: I also see this [PR](https://github.com/elastic/integrations/pull/2327) replacing all IPs. Maybe we can also change the IPs here in this PR?

**Comment by legoguy1000** [comment]
Tracking the GeoIP stuff. I will swap them out to make full use of the processor.

File: packages/aws/data_stream/route53_public_logs/agent/stream/aws-cloudwatch.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+{{/if}}
+{{/unless}}
+
+{{#unless log_group_arn}}
+{{#if log_group_name }}
+log_group_name: {{ log_group_name }}
+{{/if}}
+
+{{#if region_name }}
+region_name: {{ region_name }}
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/route53_public_logs/agent/stream/aws-cloudwatch.yml.hbs:13]: There is also `log_group_name_prefix`, `log_streams`, `log_stream_prefix` configs for the cloudwatch input.

**Comment by legoguy1000** [comment]
@kaiyan-sheng should be updated per ur comments. Also if u want to merge #2323 first then I can just update this to match how you did the closest cloud watch input

**Comment by legoguy1000** [comment]
@kaiyan-sheng When u get a chance, could you review?

**Review by kaiyan-sheng — APPROVED** [review]
@legoguy1000 Sorry for the delay!! Let's merge your PR first and then I can work on mine later!

**Comment by kaiyan-sheng** [comment]
@legoguy1000 Could you rerun the test for test-route53.log please? Thanks!!

**Comment by legoguy1000** [comment]
> 

Should be good now.

**Comment by kaiyan-sheng** [comment]
Hmm seems like the expected file is still a little off:
```
test case failed: Expected results are different from actual ones:  {
     "expected": [
         {
             "@timestamp": "2017-12-13T08:16:02.130Z",
             "aws": {
                 "route53": {
                     "edge_location": "FRA6",
                     "hosted_zone_id": "Z123412341234"
                 }
             },
             "cloud": {
                 "provider": "aws"
             },
             "dns": {
                 "question": {
                     "name": "example.com",
                     "registered_domain": "example.com",
                     "top_level_domain": "com",
                     "type": "A"
                 },
                 "response_code": "NOERROR"
             },
             "ecs": {
                 "version": "1.12.0"
             },
             "event": {
                 "category": [
                     "network"
                 ],
                 "kind": "event",
                 "original": "1.0 2017-12-13T08:16:02.130Z Z123412341234 example.com A NOERROR UDP FRA6 89.160.20.112 -",
                 "outcome": "success",
                 "type": [
                     "protocol"
                 ]
             },
             "network": {
                 "iana_number": "17",
                 "protocol": "dns",
                 "transport": "udp",
                 "type": "ipv4"
             },
             "related": {
                 "hosts": [
                     "example.com"
                 ],
                 "ip": [
                     "89.160.20.112"
                 ]
             },
             "source": {
                 "address": "89.160.20.112",
                 "as": {
                     "number": 29518,
                     "organization": {
                         "name": "Bredband2 AB"
                     }
                 },
                 "geo": {
-                    "city_name": "Tumba",
+                    "city_name": "Linköping",
                     "continent_name": "Europe",
                     "country_iso_code": "SE",
                     "country_name": "Sweden",
                     "location": {
-                        "lat": 59.2,
-                        "lon": 17.8167
+                        "lat": 58.4167,
+                        "lon": 15.6167
                     },
-                    "region_iso_code": "SE-AB",
-                    "region_name": "Stockholm"
+                    "region_iso_code": "SE-E",
+                    "region_name": "Östergötland County"
                 },
                 "ip": "89.160.20.112"
             },
             "tags": [
                 "preserve_original_event"
             ]
         },
         {
             "@timestamp": "2017-12-13T08:15:50.235Z",
             "aws": {
                 "route53": {
                     "edge_location": "IAD12",
                     "edns_client_subnet": "192.168.222.0/24",
                     "hosted_zone_id": "Z123412341234"
                 }
             },
             "cloud": {
                 "provider": "aws"
             },
             "dns": {
                 "question": {
                     "name": "example.com",
                     "registered_domain": "example.com",
                     "top_level_domain": "com",
                     "type": "AAAA"
                 },
                 "response_code": "NOERROR"
             },
             "ecs": {
                 "version": "1.12.0"
             },
             "event": {
                 "category": [
                     "network"
                 ],
                 "kind": "event",
                 "original": "1.0 2017-12-13T08:15:50.235Z Z123412341234 example.com AAAA NOERROR TCP IAD12 89.160.20.112 192.168.222.0/24",
                 "outcome": "success",
                 "type": [
                     "protocol"
                 ]
             },
             "network": {
                 "iana_number": "6",
                 "protocol": "dns",
                 "transport": "tcp",
                 "type": "ipv4"
             },
             "related": {
                 "hosts": [
                     "example.com"
                 ],
                 "ip": [
                     "89.160.20.112"
                 ]
             },
             "source": {
                 "address": "89.160.20.112",
                 "as": {
                     "number": 29518,
                     "organization": {
                         "name": "Bredband2 AB"
                     }
                 },
                 "geo": {
-                    "city_name": "Tumba",
+                    "city_name": "Linköping",
                     "continent_name": "Europe",
                     "country_iso_code": "SE",
                     "country_name": "Sweden",
                     "location": {
-                        "lat": 59.2,
-                        "lon": 17.8167
+                        "lat": 58.4167,
+                        "lon": 15.6167
                     },
-                    "region_iso_code": "SE-AB",
-                    "region_name": "Stockholm"
+                    "region_iso_code": "SE-E",
+                    "region_name": "Östergötland County"
                 },
                 "ip": "89.160.20.112"
             },
             "tags": [
                 "preserve_original_event"
             ]
         },
         {
             "@timestamp": "2017-12-13T08:16:03.983Z",
             "aws": {
                 "route53": {
                     "edge_location": "FRA6",
                     "edns_client_subnet": "2001:db8:abcd::/48",
                     "hosted_zone_id": "Z123412341234"
                 }
             },
             "cloud": {
                 "provider": "aws"
             },
             "dns": {
                 "question": {
                     "name": "example.com",
                     "registered_domain": "example.com",
                     "top_level_domain": "com",
                     "type": "ANY"
                 },
                 "response_code": "NOERROR"
             },
             "ecs": {
                 "version": "1.12.0"
             },
             "event": {
                 "category": [
                     "network"
                 ],
                 "kind": "event",
                 "original": "1.0 2017-12-13T08:16:03.983Z Z123412341234 example.com ANY NOERROR UDP FRA6 2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6 2001:db8:abcd::/48",
                 "outcome": "success",
                 "type": [
                     "protocol"
                 ]
             },
             "network": {
                 "iana_number": "17",
                 "protocol": "dns",
                 "transport": "udp",
                 "type": "ipv6"
             },
             "related": {
                 "hosts": [
                     "example.com"
                 ],
                 "ip": [
                     "2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6"
                 ]
             },
             "source": {
                 "address": "2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6",
-                "as": {
-                    "number": 62121,
-                    "organization": {
-                        "name": "Christian Ebsen ApS"
-                    }
-                },
                 "geo": {
                     "continent_name": "Europe",
-                    "country_iso_code": "DK",
-                    "country_name": "Denmark",
+                    "country_iso_code": "NO",
+                    "country_name": "Norway",
                     "location": {
-                        "lat": 56,
+                        "lat": 62,
                         "lon": 10
                     }
                 },
                 "ip": "2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6"
             },
             "tags": [
                 "preserve_original_event"
             ]
         },
         {
             "@timestamp": "2017-12-13T08:15:50.342Z",
             "aws": {
                 "route53": {
                     "edge_location": "IAD12",
                     "edns_client_subnet": "192.168.111.0/24",
                     "hosted_zone_id": "Z123412341234"
                 }
             },
             "cloud": {
                 "provider": "aws"
             },
             "dns": {
                 "question": {
                     "name": "bad.example.com",
                     "registered_domain": "example.com",
                     "subdomain": "bad",
                     "top_level_domain": "com",
                     "type": "A"
                 },
                 "response_code": "NXDOMAIN"
             },
             "ecs": {
                 "version": "1.12.0"
             },
             "event": {
                 "category": [
                     "network"
                 ],
                 "kind": "event",
                 "original": "1.0 2017-12-13T08:15:50.342Z Z123412341234 bad.example.com A NXDOMAIN UDP IAD12 89.160.20.112 192.168.111.0/24",
                 "outcome": "failure",
                 "type": [
                     "protocol"
                 ]
             },
             "network": {
                 "iana_number": "17",
                 "protocol": "dns",
                 "transport": "udp",
                 "type": "ipv4"
             },
             "related": {
                 "hosts": [
                     "bad.example.com"
                 ],
                 "ip": [
                     "89.160.20.112"
                 ]
             },
             "source": {
                 "address": "89.160.20.112",
                 "as": {
                     "number": 29518,
                     "organization": {
                         "name": "Bredband2 AB"
                     }
                 },
                 "geo": {
-                    "city_name": "Tumba",
+                    "city_name": "Linköping",
                     "continent_name": "Europe",
                     "country_iso_code": "SE",
                     "country_name": "Sweden",
                     "location": {
-                        "lat": 59.2,
-                        "lon": 17.8167
+                        "lat": 58.4167,
+                        "lon": 15.6167
                     },
-                    "region_iso_code": "SE-AB",
-                    "region_name": "Stockholm"
+                    "region_iso_code": "SE-E",
+                    "region_name": "Östergötland County"
                 },
                 "ip": "89.160.20.112"
             },
             "tags": [
                 "preserve_original_event"
             ]
         },
         {
             "@timestamp": "2017-12-13T08:16:05.744Z",
             "aws": {
                 "route53": {
                     "edge_location": "JFK5",
                     "hosted_zone_id": "Z123412341234"
                 }
             },
             "cloud": {
                 "provider": "aws"
             },
             "dns": {
                 "question": {
                     "name": "txt.example.com",
                     "registered_domain": "example.com",
                     "subdomain": "txt",
                     "top_level_domain": "com",
                     "type": "TXT"
                 },
                 "response_code": "NOERROR"
             },
             "ecs": {
                 "version": "1.12.0"
             },
             "event": {
                 "category": [
                     "network"
                 ],
                 "kind": "event",
                 "original": "1.0 2017-12-13T08:16:05.744Z Z123412341234 txt.example.com TXT NOERROR UDP JFK5 89.160.20.112 -",
                 "outcome": "success",
                 "type": [
                     "protocol"
                 ]
             },
             "network": {
                 "iana_number": "17",
                 "protocol": "dns",
                 "transport": "udp",
                 "type": "ipv4"
             },
             "related": {
                 "hosts": [
                     "txt.example.com"
                 ],
                 "ip": [
                     "89.160.20.112"
                 ]
             },
             "source": {
                 "address": "89.160.20.112",
                 "as": {
                     "number": 29518,
                     "organization": {
                         "name": "Bredband2 AB"
                     }
                 },
                 "geo": {
-                    "city_name": "Tumba",
+                    "city_name": "Linköping",
                     "continent_name": "Europe",
                     "country_iso_code": "SE",
                     "country_name": "Sweden",
                     "location": {
-                        "lat": 59.2,
-                        "lon": 17.8167
+                        "lat": 58.4167,
+                        "lon": 15.6167
                     },
-                    "region_iso_code": "SE-AB",
-                    "region_name": "Stockholm"
+                    "region_iso_code": "SE-E",
+                    "region_name": "Östergötland County"
                 },
                 "ip": "89.160.20.112"
             },
             "tags": [
                 "preserve_original_event"
             ]
         }
     ]
 } 
```

**Comment by legoguy1000** [comment]
Could it be a stack version mismatch? I think I generated the data with a 7.16.x stack but it looks like Jenkins spun up 7.15. I can see if that makes a difference.

**Comment by kaiyan-sheng** [comment]
@legoguy1000 I think that's the problem hmm

**Comment by legoguy1000** [comment]
7.15.0 stack didn't produce any different data for my locally run test, trying 8.0.0-SNAPSHOT.

**Comment by legoguy1000** [comment]
> 7.15.0 stack didn't produce any different data for my locally run test, trying 8.0.0-SNAPSHOT.

Looks like I just needed to run `elastic-package stack update --version 7.16.2` and now the geoIP value are consistent between 7.x and 8.0

**Pull request merged**